pkg/inventory/inventory-client.go

```diff
@@ -441,24 +448,28 @@ func (cic *ClusterInventoryClient) updateStatus(obj *unstructured.Unstructured,
 	if !found {
 		return nil
 	}
-
-	klog.V(4).Infof("update inventory status")
 	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
+	if err != nil {
+		return nil
+	}
+	hasStatus, err := cic.hasSubResource(obj.GetAPIVersion(), mapping.Resource.Resource, "status")
 	if err != nil {
 		return err
 	}
+	if !hasStatus {
+		klog.V(4).Infof("skip updating inventory status")
+		return nil
+	}
+
+	klog.V(4).Infof("update inventory status")
 	resource := cic.dc.Resource(mapping.Resource).Namespace(obj.GetNamespace())
 	meta := metav1.TypeMeta{
 		Kind:       obj.GetKind(),
 		APIVersion: obj.GetAPIVersion(),
 	}
-	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta}, "status")
+	liveObj, err := resource.Get(context.TODO(), obj.GetName(), metav1.GetOptions{TypeMeta: meta})
```

Comment:
Can we skip this GET call? Or is it necessary in order to do an update with the latest spec & metadata? I think we have the original desired spec in `obj`, right?

Reply:
Also, was the "status" subresource dropped here intentionally?

Reply:
This is necessary since the update needs to be with latest spec and metadata. The `obj` may not have the latest metadata.

Reply:
Yes, the "status" was dropped intentionally. It was originally included to check if the resource has the "status" subresource. Now we use the discovery client to check this. The "status" here can be dropped.

Reply:
Do subresources have distinct metadata?
If you write metadata to a subresource does it change the parent resource's metadata?

Reply:
kernel team says CRD subresources drop/ignore metadata changes.
So we don't need to get new metadata. The only meta that matters are the ones used to identify the resource.

Reply:
ConfigMaps don't have subresources, so it doesn't matter for that use case.

Reply:
The Spec is also being dropped/ignored when you update the status, so we don't need to update that either.

I think the whole GET can be skipped.